samples/sdl_gamecontroller/main.c

```diff
@@ -0,0 +1,108 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/video.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+int main() {
+  XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+  int gameControllerIndex = 0;
+  assert(SDL_IsGameController(gameControllerIndex));
+  SDL_GameController *gameController = SDL_GameControllerOpen(gameControllerIndex);
+  if (gameController == NULL) {
+    debugPrint("Could not open GameController %i, exiting...", gameControllerIndex);
+    SDL_QuitSubSystem(SDL_INIT_GAMECONTROLLER);
+    XSleep(2000);
+    return -1;
+  }
+
+  debugPrint("GameController name: %s\n", SDL_GameControllerName(gameController));
+  debugPrint("GameController Map : %s\n", SDL_GameControllerMapping(gameController));
+  debugPrint("\n");
+
+  while (true) {
+    // Fetch current GameController state
+    SDL_GameControllerUpdate();
+    debugClearScreen();
```

Comment:
This hides any information printed before this loop; could there be a better way, where you only clear one half of the screen or re-print other messages?

The sample starts, prints stuff, immediately deletes it. It's okay, but then it shouldn't print stuff before the loop.

Reply:
I couldn't find such a cleaning procedure, went the "remove unreadable messages" route instead.